registry/extension/oras/referrers.go

```diff
@@ -41,6 +58,37 @@ func (h *referrersHandler) getReferrers(w http.ResponseWriter, r *http.Request)
 		referrers = []orasartifacts.Descriptor{}
 	}
 
+	// only consider pagination if # of referrers is greater than page size
+	if len(referrers) > nPage {
+		startIndex := 0
+		if nextToken != "" {
+			for i, ref := range referrers {
+				if ref.Digest.String() == nextToken {
+					startIndex = i + 1
+				}
+			}
+			// if matching referrer not found
+			if startIndex == 0 {
+				h.extContext.Errors = append(h.extContext.Errors, v2.ErrorCodeReferrerNotFound.WithDetail("matching referrer with digest in nextToken not found"))
```

Comment:
If a nextToken is provided but we are unable to resume pagination due to the corresponding previous manifests being not found anymore, returning a 404 would indicate a non-retryable condition to the client. Instead, how about just start over from page 1 at this point?

Reply:
that makes sense. I'll switch behavior to that